### PR TITLE
Import keycloak operator controller into gitops-operator

### PR DESCRIPTION
**Pull request by iam-veeramalla**
Signed-off-by: iam-veeramalla <abhishek.veeramalla@gmail.com>

Draft PR
Current Issue:
         1. RHSSO is a maintainance fork of Key lock operator version 9 , which is very old and using old version of k8s libraries in go.mod , it's conflicting with the version used in GitOps operator.
       2. There seems to be bug registering the keycloakclient controller latest version(from source).

### CODE REVIEW

File: deploy/olm-catalog/gitops-operator/manifests/gitops-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -231,6 +246,23 @@ spec:
...
+          - keycloakrealms/status
+          - keycloakrealms/finalizers
+          - keycloakclients
+          - keycloakclients/status
+          - keycloakclients/finalizers
+          verbs:
+          - create
+          - get
+          - list
+          - watch
```
Comment (sbose78) [inline deploy/olm-catalog/gitops-operator/manifests/gitops-operator.clusterserviceversion.yaml:265]: Do we need to watch?

File: deploy/olm-catalog/gitops-operator/manifests/keycloakclients.keycloak.org.crd.yaml [hunk 2]
```diff
@@ -0,0 +1,268 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: keycloakclients.keycloak.org
```
Comment (sbose78) [inline deploy/olm-catalog/gitops-operator/manifests/keycloakclients.keycloak.org.crd.yaml:4]: No, we are not shipping the CRDs for keycloak.

File: pkg/controller/add_keycloak.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+import (
+	keycloak "github.com/keycloak/keycloak-operator/pkg/controller/keycloak"
+	keycloakclient "github.com/keycloak/keycloak-operator/pkg/controller/keycloakclient"
+	keycloakrealm "github.com/keycloak/keycloak-operator/pkg/controller/keycloakrealm"
+)
+
+func init() {
+	// AddToManagerFuncs is a list of functions to create controllers and add them to a manager.
+	AddToManagerFuncs = append(AddToManagerFuncs, keycloak.Add)
```
Comment (sbose78) [inline pkg/controller/add_keycloak.go:11]: No, we are not importing keycloak controllers.

**Review by sbose78 — CHANGES_REQUESTED** [review]

**Comment by iam-veeramalla** [comment]
Closing the PR. This PR was initially created with an intent to install RHSSO using keycloak-operator.

We came up with a new design to install RHSSO using openshift-templates. This feature is developed and merged in upstream to support SSO using keycloak for every Argo CD instance.
https://github.com/argoproj-labs/argocd-operator/commit/f11b440c49df0ea3ead9d33d3484c7160b1397ef

**Pull request closed without merging**